partitions.go

```diff
@@ -203,3 +203,60 @@ func (m *partitionMap) Info() map[string][]int32 {
 	}
 	return info
 }
+
+// Diffs map with current partitionMap, returns additions and removals
+func Diff(current map[string][]int32, new map[string][]int32) (map[string][]int32, map[string][]int32) {
```

Comment:
Does `Diff` really need to be a public function? Please protect, a unit test would be good too